modules/javafx.graphics/src/main/java/com/sun/javafx/tk/quantum/GlassWindowEventHandler.java

```diff
@@ -105,15 +107,7 @@ public Void get() {
                 float outScaleX = window.getOutputScaleX();
                 float outScaleY = window.getOutputScaleY();
                 stage.stageListener.changedScale(outScaleX, outScaleY);
-                // We need to sync the new scales for painting
-                QuantumToolkit.runWithRenderLock(() -> {
-                    GlassScene scene = stage.getScene();
-                    if (scene != null) {
-                        scene.entireSceneNeedsRepaint();
-                        scene.updateSceneState();
-                    }
-                    return null;
-                });
+                forceRepaint();
```

Comment:
(github diff is confusing, this code is for `WindowEvent.RESCALE` case)

I suspect this might also fix an issue that I encountered in one of my applications which uses Canvas-based text editor.  After the macOS is woken up after a sleep with an external monitor attached, the text on retina looked blurred.  I could not get a reproducer in time, but I think it was because while I put it to sleep, the scene got re-rendered on the external monitor going from scale 2 to scale 1, but when woken up, the scene was correctly re-rendered for retina but the canvas was not resized, requiring it to be scaled.

I'll try to test this scenario, but it may take a few days.

Reply:
actually, the reason I bumped the reviewer count is to let you add the issue and make sure we get the testing results for linux (which we got) and our CI run (which we did).

I may not be able to test my failing scenario before you integrate anyway.  We are good to go.